tests/TestCase/Database/QueryTest.php

```diff
@@ -4697,4 +4669,75 @@ public function assertQuotedQuery($pattern, $query, $optional = false)
         $pattern = str_replace('>', '[`"\]]' . $optional, $pattern);
         $this->assertRegExp('#' . $pattern . '#', $query);
     }
+
+    /**
+     * Test that calling fetchAssoc return an associated array.
+     * @return void
+     * @throws \Exception
+     */
+    public function testFetchAssoc()
+    {
+        $this->loadFixtures('Profiles');
+        $query = new Query($this->connection);
+        $results = $query
+            ->select([
+                'id',
+                'user_id',
+                'is_active'
+            ])
+            ->from('profiles')
+            ->limit(1)
+            ->execute()
+            ->fetchAssoc();
+        $this->assertSame(['id' => '1', 'user_id' => '1', 'is_active' => '0'], $results);
+    }
+
+    /**
+     * Test that fetchColumn() will return the correct value at $position.
+     * @throws \Exception
+     * @return void
+     */
+    public function testFetchColumn()
+    {
+        $this->loadFixtures('Profiles');
+        $query = new Query($this->connection);
+        $query
+            ->select([
+                'id',
+                'user_id',
+                'is_active'
+            ])
+            ->from('profiles')
+            ->where(['id' => 2])
+            ->limit(1);
+        $statement = $query->execute();
+        $results = $statement->fetchColumn(0);
+        $this->assertSame('2', $results[0]);
```

Comment:
Why is fetchColumn() returning an array? Shouldn't it return the scalar value in the column?

Reply:
It's not actually returning an array, it's returning a string with 1 character and passing. I updated test so it is correct now. I also corrected the code got  rid of the last 2 places where the FETCH_TYPE_OBJ  constants were located. 

I wonder if I should update the fetch/fetchall docblocks to show the use of the constants instead of the strings ('num', 'assoc') or just leave it as is? Does anyone have a preference?

Reply:
> I wonder if I should update the fetch/fetchall docblocks to show the use of the constants instead of the strings ('num', 'assoc') or just leave it as is?

Constants are better imo, as they fail hard on copy/paste typo errors.